src/filters/factory.rs

```diff
@@ -24,6 +24,23 @@ use crate::{
 /// An owned pointer to a dynamic [`FilterFactory`] instance.
 pub type DynFilterFactory = Box<dyn FilterFactory>;
 
+/// The value returned by [`FilterFactory::create_filter`].
+pub struct CreatedFilter {
+    /// The configuration used to create the filter.
+    pub config: serde_json::Value,
+    /// The created filter.
+    pub filter: Box<dyn Filter>,
+}
+
+impl From<(serde_json::Value, Box<dyn Filter>)> for CreatedFilter {
```

Comment:
I'm not sure about this `From` implementation, if we add new things to `CreatedFilter` we'll have to change it, and it feels odd that `(Value, Box)` would work but `(Box, Value)` won't. What would you think about having a `new` method instead?